### PR TITLE
Polygott Optimizations

### PR DESCRIPTION
**Pull request by leon332157**
This aims to provide a faster loading speed and a more efficent package system with the migration to polygott. The code has been tested in my forked repl.

### CODE REVIEW

**Comment by leon332157** [comment]
Also Since i committed on repl.it it shows mat committed but it's actually me.

**Comment by Codemonkey51** [comment]
@leon332157 please dynamically set path (by appending)

**Comment by leon332157** [comment]
> 
> 
> @leon332157 please dynamically set path (by appending)

What path.

**Comment by Codemonkey51** [comment]
> > @leon332157 please dynamically set path (by appending)
> 
> What path.

The `PATH` environment var you set in main.sh (also pls no commit lock files)

**Comment by leon332157** [comment]
The PATH is set so it works with the polygott repl. The lockfile is needed because it will reduce the time to install packages by about 2 minutes.

**Comment by Codemonkey51** [comment]
> The PATH is set so it works with the polygott repl. The lockfile is needed because it will reduce the time to install packages by about 2 minutes.

Oh ok, but still PATH has some parts set by default (by replit) and it's better to just add more to path with `export PATH=$PATH:"<new_entries>"`

**Comment by leon332157** [comment]
Turns out I don't need PATH. So I just removed it. Could you guys test this PR by forking the main repl and running it? Thanks.

**Comment by Codemonkey51** [comment]
@leon332157 im not saying remove those entries, but just append them not hard set

**Comment by Codemonkey51** [comment]
> Turns out I don't need PATH. So I just removed it. Could you guys test this PR by forking the main repl and running it? Thanks.

Lmao someone has to test it locally, also I got to go

**Comment by leon332157** [comment]
@Codemonkey51 You don't need to since this doesn't involve anything code-related. It's purely polygott.

File: .replit [hunk 1]
```diff
@@ -1,2 +1,2 @@
-language = "python3"
```
Comment (Codemonkey51) [inline .replit:1]: One thing, keep the language="python3" pls

**Review by Codemonkey51 — CHANGES_REQUESTED** [review]

File: .replit (hunk 1 above)
Comment (leon332157) [reply .replit:1]: That's not needed anymore.

Comment (Codemonkey51) [reply .replit:1]: Oh ok 👍

**Review by Codemonkey51 — APPROVED** [review]

**Comment by Codemonkey51** [comment]
> Turns out I don't need PATH. So I just removed it. Could you guys test this PR by forking the main repl and running it? Thanks.

Well I can just pull it in, run it then if it doesn't work revert to last commit b4 and make new or to fix?

**Comment by leon332157** [comment]
Don’t merge until 2 people have tested this with a forked repl. Thanks.

**Comment by Codemonkey51** [comment]
> Don’t merge until 2 people have tested this with a forked repl. Thanks.

Ok, sry I can't do that rn (should I request changes so someone doesn't accidentally merge?)

**Comment by Codemonkey51** [comment]
That also works

**Comment by leon332157** [comment]
I’ll just wait for Prussia or mat to test it. Don’t worry for now.

**Comment by Codemonkey51** [comment]
Imma go now

**Comment by Codemonkey51** [comment]
Hey just popping back for like 1 minute, storing .venv in the repl folder inflates repl size by a lot, so could you maybe store this in like ~/.venv or something

**Comment by Codemonkey51** [comment]
Also this hard codes the folder, making local hosting annoying

**Review by jetstream0 — COMMENTED** [review]
https://repl.it/@rediar/Repldex-1 forked and ran it in repl.it, seems like it worked properly (packages installed, main.py ran)

**Comment by jetstream0** [comment]
Seems like you want another person to test this, so won't approve on github so no one accidentally merges, but I approve

**Comment by Codemonkey51** [comment]
@leon332157 maybe should a file be made like /etc/repldex/deps.lock* so if that exists (and it is the same as pyproject.toml) it doesn't re-install but if not it will install (so repl doesn't run poetry on every restart, just on first container run)

*it has to be a file not in /home/runner/<repl-slug> (the /etc/repldex thing also works on self hosts, as long as they are on linux)

**Comment by Codemonkey51** [comment]
Wait.... repls can't edit etc/ I'll think of a different place :/

**Comment by leon332157** [comment]
I’m not sure how that will work since we created venv outside the project directory. But I could try detecting if packages are installed.

**Comment by leon332157** [comment]
So if there's a case where even thought venv exist, packages are not installed, removing venv will install the packages.

**Comment by leon332157** [comment]
@Codemonkey51 Any change you can test this or approve the review, I think this should be merged today.

File: main.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+export PYTHONPATH='/usr/local/lib/python3.8/site-packages'
+export POETRY_VIRTUALENVS_PATH='/home/runner/venv'
+export POETRY_VIRTUALENVS_CREATE='true'
+/usr/local/bin/python3 -m pip show poetry
+if [ $? -ne 0 ]; then
+  echo "Installing poetry"
+  /usr/local/bin/python3 -m pip install poetry --user
+fi
+clear
+if [ ! -d "/home/runner/venv" ];then
```
Comment (Codemonkey51) [inline main.sh]: Use $POETRY_VIRTUALENVS_PATH

File: main.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+unset PYTHONPATH
+unset VIRTUAL_ENV #/opt/virtualenvs/python3
+export PYTHONPATH='/usr/local/lib/python3.8/site-packages'
+export POETRY_VIRTUALENVS_PATH='/home/runner/venv'
```
Comment (Codemonkey51) [inline main.sh]: use /home/runner/.venv

**Review by Codemonkey51 — CHANGES_REQUESTED** [review]
Just 2 quick changes needed

**Review by Codemonkey51 — APPROVED** [review]

**Comment by leon332157** [comment]
keeping the branch just incase something breaks

**Pull request merged**